### PR TITLE
Renew dhcp lease

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: src/tcpip/tcpip.c [hunk 1]
```diff
@@ -176,10 +176,7 @@ static void onstatechange(struct mg_tcpip_if *ifp) {
   if (ifp->state == MG_TCPIP_STATE_READY) {
     MG_INFO(("READY, IP: %M", mg_print_ip4, &ifp->ip));
     MG_INFO(("       GW: %M", mg_print_ip4, &ifp->gw));
-    if (ifp->lease_expire > ifp->now) {
-      MG_INFO(
-          ("       Lease: %lld sec", (ifp->lease_expire - ifp->now) / 1000));
-    }
+    MG_INFO(("      MAC: %M", mg_print_mac, &ifp->mac));
```
Comment (scaprile) [inline src/tcpip/tcpip.c:179]: We are printing the MAC only on those examples where the user would not know it (we generate it).
This is not useful info when the user has set the MAC and makes us print it twice when it is.

File: src/tcpip/tcpip.c [hunk 2]
```diff
@@ -347,11 +346,14 @@ static void rx_dhcp_client(struct mg_tcpip_if *ifp, struct pkt *pkt) {
...
       memcpy(&lease, p + 2, sizeof(lease));
-      ifp->lease_expire = ifp->now + mg_ntohl(lease) * 1000;
+      lease = mg_ntohl(lease);
+      ifp->lease_expire = ifp->now + lease * 1000;
+      MG_INFO(("Lease: %u sec (%lld)", lease, ifp->lease_expire / 1000));
     }
     p += p[1] + 2;
   }
   if (ip && mask && gw && ifp->ip == 0) {
+    MG_INFO(("OO"));
```
Comment (scaprile) [inline src/tcpip/tcpip.c:356]: I think this one is a leftover

**Review by scaprile — COMMENTED** [review]
This should work when the boards are always on. Let's go for it !
I fail to see the problem when the board starts with new firmware and asks for an IP. It should be given a new one if it expired; and if given one that just expires in seconds, it will get a new one on next reboot.

File: src/tcpip/tcpip.c (hunk 1 above)
Comment (cpq) [reply src/tcpip/tcpip.c:179]: It actually might be useful to see logs, to spot conflicts or outright bad MAC settings

File: src/tcpip/tcpip.c (hunk 2 above)
Comment (cpq) [reply src/tcpip/tcpip.c:356]: It is !

**Pull request merged**